src/univariate/continuous/maximum.jl

```diff
@@ -0,0 +1,25 @@
+"""
+The maximum of n iid random variables with continuous univariate distribution
+"""
+struct Maximum{D<:ContinuousUnivariateDistribution} <: ContinuousUnivariateDistribution
+    dist::D
+    n::Int
+    Maximum{D}(dist, n) where {D<:ContinuousUnivariateDistribution} = new{D}(dist, n)
+end
+
+function Maximum(dist::D, n::Integer; check_args::Bool=true) where {D <: ContinuousUnivariateDistribution}
+    @check_args Maximum (n, n >= one(n))
+    return Maximum{D}(dist, n)
+end
+
+rand(rng::AbstractRNG, d::Maximum{D}) where {D} = maximum([rand(rng, d.dist) for _ in 1:d.n])
+
+#### Evaluation
+
+cdf(d::Maximum{D}, x::Real) where {D} = cdf(d.dist, x)^d.n
+pdf(d::Maximum{D}, x::Real) where {D} = d.n*pdf(d.dist, x)*cdf(d.dist, x)^(d.n-1)
+logpdf(d::Maximum{D}, x::Real) where {D} = log(d.n)+logpdf(d.dist, x)+(d.n-1)*logcdf(d.dist, x)
+quantile(d::Maximum{D}, q::Real) where {D} = quantile(d.dist, q^(1/d.n))
```

Comment:
No need for `D`, and we can use `//` to avoid unnecessarily promoting to a `Float64`. Also, I'd have to think more about how if this needs to be changed to handle the discrete case.
```suggestion
quantile(d::Maximum, q::Real) = quantile(d.dist, q^(1//d.n))
```

Reply:
It seems nothing special needs to be done, see https://github.com/JuliaStats/Distributions.jl/pull/1655#issuecomment-1407654229.